src/DataModel/IApiDocumentation.ts

```diff
@@ -0,0 +1,9 @@
+import {IMetadataDescription} from "./IMetadataDescription";
+import {IData} from "./IData";
+
+export interface IApiDocumentation extends IMetadataDescription
+{
+    entryPoint: string;
+
+    getEntryPoint(): Promise<IData>;
```

Comment:
From my experience in implementing hypermedia clients with `System.Net.HttpClient`, I find that having a central infrastructural and generic `Client` class that deals with HTTP communication, error handling, serialization, deserialization, etc., and then passing it around to the different specifics of the library works well.

Equivalently, I think it's wise to make `ApiDocumentation` rely on `Client` to do the heavy lifting and then just applying the specifics within the `ApiDocumentation` class.